scripts/build_locally.py

```diff
@@ -65,6 +67,23 @@ def run(
         ]
     if cmake_opts:
         cmake_args += cmake_opts.split()
+    if target_cuda is not None:
+        if not target_cuda.strip():
+            raise ValueError(
+                "--target-cuda can not be an empty string. "
+                "Use --target-cuda=<arch> or --target-cuda"
+            )
+        cmake_args += [
+            f"-DDPCTL_TARGET_CUDA={target_cuda}",
+        ]
```

Comment:
what would happen if `-DDPCTL_TARGET_CUDA={target_cuda}` and `--target-cuda` both happen to be specified?

Reply:
@ndgrigorian good question!
When both are passed, CMake usually uses the first value.
To avoid confusion, I have added a check that prevents using both `--target-cuda/--target-hip` and the matching CMake option in `--cmake-opts`
Thank you